Modules/TPC/src/Clusters.cxx

```diff
@@ -69,39 +66,64 @@ void Clusters::initialize(o2::framework::InitContext& /*ctx*/)
-
-  for (auto& histPtr : mHistoVector) {
-    auto hist = histPtr.get();
-    getObjectsManager()->startPublishing(hist);
-    getObjectsManager()->addMetadata(hist->GetName(), "custom", "34");
+  mNClustersCanvasVec.emplace_back(std::make_unique<TCanvas>("c_Sides_N_Clusters"));
+  mNClustersCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_N_Clusters_1D"));
+  mNClustersCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_N_Clusters_2D"));
+
+  mQMaxCanvasVec.emplace_back(std::make_unique<TCanvas>("c_Sides_Q_Max"));
+  mQMaxCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Q_Max_1D"));
+  mQMaxCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Q_Max_2D"));
+
+  mQTotCanvasVec.emplace_back(std::make_unique<TCanvas>("c_Sides_Q_Tot"));
+  mQTotCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Q_Tot_1D"));
+  mQTotCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Q_Tot_2D"));
+
+  mSigmaTimeCanvasVec.emplace_back(std::make_unique<TCanvas>("c_Sides_Sigma_Time"));
+  mSigmaTimeCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Sigma_Time_1D"));
+  mSigmaTimeCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Sigma_Time_2D"));
+
+  mSigmaPadCanvasVec.emplace_back(std::make_unique<TCanvas>("c_Sides_Sigma_Pad"));
+  mSigmaPadCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Sigma_Pad_1D"));
+  mSigmaPadCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Sigma_Pad_2D"));
+
+  mTimeBinCanvasVec.emplace_back(std::make_unique<TCanvas>("c_Sides_Time_Bin"));
+  mTimeBinCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Time_Bin_1D"));
+  mTimeBinCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Time_Bin_2D"));
+
+  for (auto& canvasPtr : mNClustersCanvasVec) {
+    auto canvas = canvasPtr.get();
+    getObjectsManager()->startPublishing(canvas);
+    getObjectsManager()->addMetadata(canvas->GetName(), "custom", "34");
+  }
+
+  for (auto& canvasPtr : mQMaxCanvasVec) {
+    auto canvas = canvasPtr.get();
+    getObjectsManager()->startPublishing(canvas);
+    getObjectsManager()->addMetadata(canvas->GetName(), "custom", "34");
+  }
+
+  for (auto& canvasPtr : mQTotCanvasVec) {
+    auto canvas = canvasPtr.get();
+    getObjectsManager()->startPublishing(canvas);
+    getObjectsManager()->addMetadata(canvas->GetName(), "custom", "34");
+  }
+
+  for (auto& canvasPtr : mSigmaTimeCanvasVec) {
+    auto canvas = canvasPtr.get();
+    getObjectsManager()->startPublishing(canvas);
+    getObjectsManager()->addMetadata(canvas->GetName(), "custom", "34");
+  }
+
+  for (auto& canvasPtr : mSigmaPadCanvasVec) {
+    auto canvas = canvasPtr.get();
+    getObjectsManager()->startPublishing(canvas);
+    getObjectsManager()->addMetadata(canvas->GetName(), "custom", "34");
+  }
+
+  for (auto& canvasPtr : mTimeBinCanvasVec) {
+    auto canvas = canvasPtr.get();
+    getObjectsManager()->startPublishing(canvas);
+    getObjectsManager()->addMetadata(canvas->GetName(), "custom", "34");
   }
```

Comment:
This is a bit ugly since it is so much copy'n'paste. Could also be made nicer using a lamba. E.g.
```suggestion
addAndPublish = [](auto& canVec, std::vector<std::string_view> canvNames) {
  for (const auto& canvName : canvNames) {
    canvVec.emplace_back(std::make_unique<TCanvas>(canvName.data()));
    auto canvas = canVec.back().get();
    getObjectsManager()->startPublishing(canvas);
    getObjectsManager()->addMetadata(canvas->GetName(), "custom", "34");
  }
}

  addAndPublish(mNClustersCanvasVec, {"c_Sides_N_Clusters", "c_ROCs_N_Clusters_1D", "c_ROCs_N_Clusters_2D"});
  addAndPublish(mQMaxCanvasVec, {"c_Sides_Q_Max", "c_ROCs_Q_Max_1D", "c_ROCs_Q_Max_2D"});
  addAndPublish(mQTotCanvasVec, {"c_Sides_Q_Tot", "c_ROCs_Q_Tot_1D", "c_ROCs_Q_Tot_2D"});
  addAndPublish(mSigmaTimeCanvasVec, {"c_Sides_Sigma_Time", "c_ROCs_Sigma_Time_1D", "c_ROCs_Sigma_Time_2D"});
  addAndPublish(mSigmaPadCanvasVec, {"c_Sides_Sigma_Pad", "c_ROCs_Sigma_Pad_1D", "c_ROCs_Sigma_Pad_2D"});
  addAndPublish(mTimeBinCanvasVec, {"c_Sides_Time_Bin", "c_ROCs_Time_Bin_1D", "c_ROCs_Time_Bin_2D"});
```

Same can be done in `Calibration.cxx` above. perhaps also by putting the lambda in some common space.

Reply:
This one works with a few minor changes.

Reply:
What would you suggest as a common space? Just put a Utility.h in Modules/TPC/include/TPC or go even more common?

Reply:
Yes, `Utiltiy.h` in TPC sounds good.